fedimint-server/src/net/peers.rs

```diff
@@ -406,183 +280,105 @@ where
 {
     async fn state_transition_connected(
         &mut self,
-        mut connected: ConnectedPeerConnectionState<M>,
-        task_handle: &TaskHandle,
+        mut connection: AnyFramedTransport<PeerMessage<M>>,
     ) -> Option<PeerConnectionState<M>> {
         Some(tokio::select! {
             maybe_msg = self.outgoing.recv() => {
-                if let Ok(msg) = maybe_msg {
-                    self.send_message_connected(connected, PeerMessage::Message(msg)).await
-                } else {
-                    debug!(target: LOG_NET_PEER, "Exiting peer connection IO task - parent disconnected");
-                    return None;
-                }
+                self.send_message_connected(connection, PeerMessage::Message(maybe_msg.ok()?)).await
             },
-            new_connection_res = self.incoming_connections.recv() => {
-                if let Some(new_connection) = new_connection_res {
-                    debug!(target: LOG_NET_PEER, "Replacing existing connection");
-                    self.connect(new_connection, 0).await
-                } else {
-                    debug!(
-                    target: LOG_NET_PEER,
-                        "Exiting peer connection IO task - parent disconnected");
-                    return None;
-                }
+            maybe_connection = self.incoming_connections.recv() => {
+                self.connect(maybe_connection?).await
             },
-            Some(message_res) = connected.connection.next() => {
+            Some(message_res) = connection.next() => {
                 match message_res {
                     Ok(peer_message) => {
                         if let PeerMessage::Message(msg) = peer_message {
                             PEER_MESSAGES_COUNT.with_label_values(&[&self.our_id_str, &self.peer_id_str, "incoming"]).inc();
-                            if self.incoming.try_send(msg).is_err(){
-                                debug!(target: LOG_NET_PEER, "Could not relay incoming message since the channel is full");
+
+                            if self.incoming.send(msg).await.is_err(){
```

Comment:
Notable bug fix: don't drop messages if incoming queue is full. I guess we never saw this materialize because the channel size was large enough and the consumer was quick to pull messages out.